featuregate/registry.go

```diff
@@ -77,14 +105,32 @@ func (r *Registry) MustRegister(id string, stage Stage, opts ...RegisterOption)
 	return g
 }
 
+func validateID(id string) error {
+	if id == "" {
+		return fmt.Errorf("empty ID")
+	}
+
+	if !idRegexp.MatchString(id) {
+		return fmt.Errorf("invalid character(s) in ID")
+	}
+	return nil
+}
+
 // Register a Gate and return it. The returned Gate can be used to check if is enabled or not.
 func (r *Registry) Register(id string, stage Stage, opts ...RegisterOption) (*Gate, error) {
```

Comment:
Please describe in the comment the rules for the ID string.

Reply:
Done